File: app/src/main/java/b2w/com/br/olodjinha/main/BannerAdapter.java
package b2w.com.br.olodjinha.main;

import android.content.Context;
import android.content.Intent;
import android.net.Uri;
import android.support.annotation.NonNull;
import android.support.constraint.ConstraintLayout;
import android.support.v4.view.PagerAdapter;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import com.bumptech.glide.Glide;
import com.bumptech.glide.request.RequestOptions;

import java.util.List;

import b2w.com.br.olodjinha.GlideApp;
import b2w.com.br.olodjinha.R;
import b2w.com.br.olodjinha.data.api.RetrofitClient;
import b2w.com.br.olodjinha.data.models.BannerDTO;
import b2w.com.br.olodjinha.util.GlideUtil;

public class BannerAdapter extends PagerAdapter {

    private final List<BannerDTO> mBannerList;
    private final Context mContext;

    public BannerAdapter(Context context, List<BannerDTO> bannerList) {
        mBannerList = bannerList;
        mContext = context;
    }

    @Override
    public int getCount() {
        return mBannerList.size();
    }

    @Override
    public boolean isViewFromObject(@NonNull View view, @NonNull Object o) {
        return view == ((ConstraintLayout) o);
    }

    @NonNull
    @Override
    public Object instantiateItem(@NonNull ViewGroup container, int position) {
        View itemView = ((LayoutInflater) mContext.getSystemService(Context.LAYOUT_INFLATER_SERVICE))
                .inflate(R.layout.banner_item, container, false);

        ImageView mImageView = (ImageView) itemView.findViewById(R.id.imageview_banner);

        GlideUtil.setBannerImage(mContext, mImageView, mBannerList.get(position).getUrl());

        mImageView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(Intent.ACTION_VIEW,
                        Uri.parse(mBannerList.get(position).getLink()));
                mContext.startActivity(intent);
            }
        });

        container.addView(itemView);

        return itemView;
    }

    @Override
    public void destroyItem(ViewGroup container, int position, Object object) {
        container.removeView((ConstraintLayout) object);
    }
}

File: app/src/main/kotlin/br/com/rbueno/lodjinha/ui/home/banner/BannerAdapter.kt
package br.com.rbueno.lodjinha.ui.home.banner

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import androidx.viewpager.widget.PagerAdapter
import androidx.viewpager.widget.ViewPager
import br.com.rbueno.lodjinha.R
import br.com.rbueno.lodjinha.model.Banner
import br.com.rbueno.lodjinha.util.ImageLoader
import br.com.rbueno.lodjinha.util.openUrl

class BannerAdapter(private val items: Banner) : PagerAdapter() {
    override fun isViewFromObject(view: View, otherObject: Any): Boolean {
        return view == otherObject
    }

    override fun getCount() = items.bannerItem.size

    override fun instantiateItem(container: ViewGroup, position: Int): Any {
        val imageBanner = LayoutInflater.from(container.context)
            .inflate(R.layout.view_banner_item, container, false) as ImageView

        val bannerItem = items.bannerItem[position]
        ImageLoader.loadImage(bannerItem.urlImage, imageBanner)

        imageBanner.setOnClickListener {
            container.context.openUrl(bannerItem.linkUrl)
        }

        if (container is ViewPager) {
            container.addView(imageBanner)
        }

        return imageBanner
    }

    override fun destroyItem(container: ViewGroup, position: Int, view: Any) {
        if (container is ViewPager) {
            container.removeView(view as View)
        }
    }
}